### PR TITLE
[Splunk] Code improvement

### PR DESCRIPTION
**Pull request by shmuel44**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

### CODE REVIEW

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 1]
```diff
@@ -2328,30 +2333,28 @@ def splunk_job_status(service: client.Service, args: dict) -> CommandResults | N
...
-    params = demisto.params()
     if params.get('isFetch'):
-        t = datetime.utcnow() - timedelta(hours=1)
+        t = datetime.now(timezone.utc) - timedelta(hours=1)
         time = t.strftime(SPLUNK_TIME_FORMAT)
         kwargs = {'count': 1, 'earliest_time': time}
         query = params['fetchQuery']
         try:
-            if MIRROR_DIRECTION.get(params.get('mirror_direction')) and not params.get('timezone'):
+            if MIRROR_DIRECTION.get(params.get('mirror_direction', '')) and not params.get('timezone'):
```
Comment (dorschw) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2357]: why? both `None` and `''` are falsy.

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 2]
```diff
@@ -2360,7 +2363,7 @@ def test_module(service: client.Service) -> None:
                     continue
 
                 if EVENT_ID not in item:
-                    if MIRROR_DIRECTION.get(params.get('mirror_direction')):
+                    if MIRROR_DIRECTION.get(params.get('mirror_direction', '')):
```
Comment (dorschw) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2366]: why, again?

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 3]
```diff
@@ -2552,8 +2558,7 @@ def get_kv_store_config(kv_store: client.KVStoreCollection) -> str:
     readable = [f'#### configuration for {kv_store.name} store',
                 '| field name | type |',
                 '| --- | --- |']
-    for _key, val in list(keys.items()):
-        readable.append(f'| {_key} | {val} |')
+    readable.extend(f'| {_key} | {val} |' for _key, val in list(keys.items()))
```
Comment (dorschw) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2561]:
```suggestion
    readable.extend(f'| {_key} | {val} |' for _key, val in keys.items())
```

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py (hunk 1 above)
Comment (shmuel44) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2357]: Argument 1 to "get" of "dict" has incompatible type "Optional[Any]"; expected "str" mypy

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 4]
```diff
@@ -1002,16 +995,15 @@ def handle_submitted_notables(service: client.Service, incidents, cache_object:
         cache_object (Cache): The enrichment mechanism cache object
     """
     handled_notables = []
-    enrichment_timeout = arg_to_number(str(demisto.params().get('enrichment_timeout', '5')))
-    if not enrichment_timeout:
-        enrichment_timeout = 5
+    enrichment_timeout = arg_to_number(str(demisto.params().get('enrichment_timeout', '5'))) or 5
```
Comment (dorschw) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:998]: why add `or 5`?

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 5]
```diff
@@ -1298,18 +1300,22 @@ def get_modified_remote_data_command(service: client.Service, args):
...
-    modified_notable_ids = []
     remote_args = GetModifiedRemoteDataArgs(args)
     last_update_splunk_timestamp = get_last_update_in_splunk_time(remote_args.last_update)
 
-    search = '|`incident_review` ' \
-             '| eval last_modified_timestamp=_time ' \
-             '| where last_modified_timestamp>{} ' \
-             '| fields rule_id ' \
-             '| dedup rule_id'.format(last_update_splunk_timestamp)
+    search = f'|`incident_review` \
```
Comment (dorschw) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:1306]: are you sure it's identical?

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 6]
```diff
@@ -1391,28 +1400,26 @@ def create_mapping_dict(total_parsed_results, type_field):
...
     total_parsed_results = []
     search_offset = demisto.getLastRun().get('offset', 0)
 
     current_time_for_fetch = datetime.utcnow()
-    dem_params = demisto.params()
-    if demisto.get(dem_params, 'timezone'):
-        timezone = dem_params['timezone']
-        current_time_for_fetch = current_time_for_fetch + timedelta(minutes=int(timezone))
+    if demisto.get(params, 'timezone'):
+        timezone_ = params['timezone']
```
Comment (dorschw) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:1418]:
```suggestion
    if (timezone_ := params.get('timezone')):
```

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 7]
```diff
@@ -1428,17 +1435,19 @@ def get_mapping_fields_command(service: client.Service, mapper):
...
 
-    searchquery_oneshot = dem_params['fetchQuery']
+    searchquery_oneshot = params['fetchQuery']
 
-    if demisto.get(dem_params, 'extractFields'):
-        extractFields = dem_params['extractFields']
+    if demisto.get(params, 'extractFields'):
+        extractFields = params['extractFields']
         extra_raw_arr = extractFields.split(',')
         for field in extra_raw_arr:
```
Comment (dorschw) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:1443]:
```suggestion
        for field in extractFields.split(','):
```

File: .gitlab/ci/on-push.yml [hunk 8]
```diff
@@ -331,6 +331,8 @@ xsoar_server_master:
     - if: '$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/'
     - if: '$NIGHTLY'
       when: always
+      variables:
+        KUBERNETES_CPU_REQUEST: 2000m
```
[the remark this answers is not in the document]
Comment (shmuel44) [reply .gitlab/ci/on-push.yml:335]: pre commit, i del this

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 9]
```diff
@@ -1840,27 +1839,34 @@ def update_notable_events(baseurl, comment, status=None, urgency=None, owner=Non
...
-        return 4
-    elif severity == 'high':
-        return 3
-    elif severity == 'medium':
-        return 2
-    else:
-        return 1
+def severity_to_level(severity: str | None) -> int | float:
+    match severity:
+        case'informational':
```
Comment (dorschw) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:1860]:
```suggestion
        case 'informational':
```

**Pull request merged**